orca-interlink/src/main/java/com/netflix/spinnaker/orca/interlink/events/InterlinkEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.interlink.events;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+
+/** Common interface for all interlink event messages */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "eventType")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = CancelInterlinkEvent.class, name = "CANCEL"),
+  @JsonSubTypes.Type(value = PauseInterlinkEvent.class, name = "PAUSE"),
+  @JsonSubTypes.Type(value = ResumeInterlinkEvent.class, name = "RESUME"),
+  @JsonSubTypes.Type(value = DeleteInterlinkEvent.class, name = "DELETE")
+})
+public interface InterlinkEvent {
+  enum EventType {
+    CANCEL,
+    PAUSE,
+    DELETE,
+    RESUME,
+  }
+
+  @JsonIgnore
+  EventType getEventType();
+
+  Execution.ExecutionType getExecutionType();
+
+  String getExecutionId();
```

Comment:
i think it's a bit weird to have the getters here but not the backing field... maybe it would make for lombok/jackson nightmare, but i almost feel like this `InterlinkEvent` is more of a base class than an interface.